pkg/api/v1beta3/types.go

```diff
@@ -997,14 +997,16 @@ type ReplicationControllerSpec struct {
 	// Template is the object that describes the pod that will be created if
 	// insufficient replicas are detected. This takes precedence over a
 	// TemplateRef.
-	Template *PodTemplateSpec `json:"template,omitempty" description:"object that describes the pod that will be created if insufficient replicas are detected; takes precendence over templateRef"`
+	Template       *PodTemplateSpec `json:"template,omitempty" description:"object that describes the pod that will be created if insufficient replicas are detected; takes precendence over templateRef"`
+	SequenceNumber int64            `json:"sequenceNumber" description:"A token incremented with each change to the spec."`
```

Comment:
A "token" is generally an opaque cookie that represents something. Maybe this could be interpreted as representing a specific version/generation of the desired state, but I suggest that we name this field "generation" and then refer to it as a sequence number in the description.

Also, the generation should be updated when certain metadata fields are updated, as well. The "certain" qualifier is annoying, admittedly -- we don't want fields, such as resourceVersion, that are updated when status is updated to be included -- so, only the "desired-state" or "goal" fields, such as labels, annotations, and deletionTimestamp (and deletionGracePeriodSeconds, once that goes in).